src/main/java/com/artipie/rpm/Repomd.java

```diff
@@ -188,18 +187,9 @@ private CompletableSource performUpdate(
      *
      * @param repomd The temp
      * @return Completion or error signal
-     * @todo #41:30min This method and saveGzip are very similar
-     *  but implemented differently. Factor those together as
-     *  the same method based on the implementation of the saveGzip
-     *  that is closer to the rx paradigm than saveRepo.
      */
     private CompletableSource saveRepo(final Path repomd) {
-        return Completable.fromAction(
-            () -> new BlockingStorage(this.storage).save(
-                new Key.From("repodata/repomd.xml"),
-                Files.readAllBytes(repomd)
-            )
-        );
+        return this.saveGzip(repomd, "repodata/repomd.xml");
```

Comment:
@ammaratef45 Let's extract `repodata/repomd.xml` string to a static variable and remove duplication throughout the class.